packages/desktop-client/src/components/accounts/Account.js

```diff
@@ -641,13 +636,17 @@ function SelectedTransactionsButton({
             }
 
             if (scheduleId) {
-              navigate(`/schedule/edit/${scheduleId}`, {
+              pushModal(`/schedule/edit/${scheduleId}`, {
+                // Required in order to trigger re-renders after linking the transaction
+                // TODO: figure out a way how to remove this
                 locationPtr: location,
```

Comment:
I believe `pushModal` sets `locationPtr` automatically — were you seeing issues with doing that? Maybe there were some places still referencing `locationPtr` that need updating / we should revert `parent` back to `locationPtr`

Reply:
It's for this thing: https://github.com/actualbudget/actual/blob/168b38fc4befececea6af2ce0c048d96d6953ec0/packages/desktop-client/src/components/accounts/Account.js#L2083

Not a great solution IMO and something we should revisit. But I suspect this would not be a simple fix.

The issue is that we must somehow invalidate the query cache and then load fresh data in to the table when the underlying data changes (i.e. when you "link" a schedule - the transaction table should reload). But again: I suspect a proper solution here would be a lot more complex, so keeping the existing hack for now.

Reply:
→ #1256